lib/api/confighandler.go

```diff
@@ -0,0 +1,377 @@
+		c.adjustLDAP(w, r, c.cfg.LDAP())
+	})
+}
+
+func (c *configMuxBuilder) registerGUI(path string) {
+	c.HandlerFunc(http.MethodGet, path, func(w http.ResponseWriter, _ *http.Request) {
+		sendJSON(w, c.cfg.GUI())
+	})
+
+	c.HandlerFunc(http.MethodPut, path, func(w http.ResponseWriter, r *http.Request) {
+		c.adjustGUI(w, r, config.GUIConfiguration{})
+	})
+
+	c.HandlerFunc(http.MethodPatch, path, func(w http.ResponseWriter, r *http.Request) {
+		c.adjustGUI(w, r, c.cfg.GUI())
+	})
+}
+
+func (c *configMuxBuilder) adjustConfig(w http.ResponseWriter, r *http.Request) {
+	c.mut.Lock()
+	defer c.mut.Unlock()
+	cfg, err := config.ReadJSON(r.Body, c.id)
+	r.Body.Close()
+	if err != nil {
+		l.Warnln("Decoding posted config:", err)
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	if cfg.GUI.Password, err = checkGUIPassword(c.cfg.GUI().Password, cfg.GUI.Password); err != nil {
+		l.Warnln("bcrypting password:", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	waiter, err := c.cfg.Replace(cfg)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	c.finish(w, waiter)
+}
+
+func (c *configMuxBuilder) adjustFolder(w http.ResponseWriter, r *http.Request, folder config.FolderConfiguration) {
+	c.mut.Lock()
+	defer c.mut.Unlock()
+	if err := unmarshalTo(r.Body, &folder); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	waiter, err := c.cfg.SetFolder(folder)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	c.finish(w, waiter)
+}
+
+func (c *configMuxBuilder) adjustDevice(w http.ResponseWriter, r *http.Request, device config.DeviceConfiguration) {
+	c.mut.Lock()
+	defer c.mut.Unlock()
+	if err := unmarshalTo(r.Body, &device); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	waiter, err := c.cfg.SetDevice(device)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	c.finish(w, waiter)
+}
+
+func (c *configMuxBuilder) adjustOptions(w http.ResponseWriter, r *http.Request, opts config.OptionsConfiguration) {
+	c.mut.Lock()
+	defer c.mut.Unlock()
+	if err := unmarshalTo(r.Body, &opts); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	waiter, err := c.cfg.SetOptions(opts)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	c.finish(w, waiter)
+}
+
+func (c *configMuxBuilder) adjustGUI(w http.ResponseWriter, r *http.Request, gui config.GUIConfiguration) {
+	c.mut.Lock()
+	defer c.mut.Unlock()
+	err := unmarshalTo(r.Body, &gui)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	if gui.Password, err = checkGUIPassword(c.cfg.GUI().Password, gui.Password); err != nil {
+		l.Warnln("bcrypting password:", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
```

Comment:
```suggestion
	oldPassword := gui.Password        
	err := unmarshalTo(r.Body, &gui)
	if err != nil {
		http.Error(w, err.Error(), http.StatusBadRequest)
		return
	}
	if gui.Password, err = checkGUIPassword(oldPassword, gui.Password); err != nil {
		l.Warnln("bcrypting password:", err)
		http.Error(w, err.Error(), http.StatusInternalServerError)
		return
	}
```

Reply:
Eh it doesn't do the diff nicely, but what I means was just to cache the password in oldPassword instead of calling on cfg.GUI() once more